compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -102,8 +102,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 );
 
                 // Unpack `BlockAnd<BasicBlock>` into `(then_blk, else_blk)`.
-                let (then_blk, mut else_blk);
-                else_blk = unpack!(then_blk = then_and_else_blocks);
+                let (then_blk, mut else_blk) = then_and_else_blocks.unpack_to_pair();
```

Comment:
Why not just

```suggestion
                let BlockAnd(then_blk, mut else_blk) = then_and_else_blocks;
```

this would allow us to get rid of `unpack_to_pair` which essentially converts a tuple struct to a tuple.